File: Huddle.MetricWebApp/Models/Issue.cs
/*   
 *   * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.  
 *   * See LICENSE in the project root for license information.  
 */

using System;

namespace Huddle.MetricWebApp.Models
{
    public class Issue : IQueryResult
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public Category Category { get; set; }

        public int Metric { get; set; }

        public int State { get; set; }

        public DateTime StartDate { get; set; }

        public string MSTeamId { get; set; }

        public string Owner { get; set; }

        public int ActiveMetricCount { get; set; }
    }
}

File: Huddle.MetricWebApp/app/shared/models/issue.ts
/*   
 *   * Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.  
 *   * See LICENSE in the project root for license information.  
 */

import { Category } from './category';
import { State } from './state';
import { IssueState } from './issueState';
import { QueryResult } from './queryResult';

export class Issue implements QueryResult {
    id?: number;
    name?: string;
    category?: Category;
    metric?: string;
    targetGoal?: string;
    owner?: string;
    issueState?: IssueState;
    startDate?: Date;
    isMetricEditable?: boolean;
    isNameEditable?: boolean;
    activeMetricCount?: number;
}
